### PR TITLE
Docs for KVM Ingestion feature

### PR DESCRIPTION
**Pull request by kishankavala**
Added documentation for Import from External Host, Import from Local/Shared storage

----
📚 Documentation preview 📚: https://cloudstack-documentation--356.org.readthedocs.build/en/356/

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan ui

**Comment by borisstoyanov** [comment]
@blueorangutan doc-build

File: source/adminguide/virtual_machines/importing_unmanaging_vms.rst [hunk 1]
```diff
@@ -401,6 +409,170 @@ Unmanaging Instance actions
...
+   - **domainid** (An optional domain ID for the Instance. Must be used with account parameter)
+   - **projectid**
+   - **serviceofferingid**
+   - **nicnetworklist** (Map for NIC ID and corresponding Network UUID)
+   - **nicipaddresslist** (Map for NIC ID and corresponding IP address)
+   - **datadiskofferinglist** (Map for data disk ID and corresponding disk offering UUID)
+   - **details** (Map for Instance details)
+   - **forced** (If true, an Instance is imported despite some of its NIC's MAC addresses being already present)
+
+.. note:: The `forced` parameter is false by default and prevents importing an Instance which has a NIC containing a
```
Comment (kiranchavala) [inline source/adminguide/virtual_machines/importing_unmanaging_vms.rst:507]: The formatting is incorrect 
Screenshot 2023-11-13 at 2 36 29 PM

File: source/adminguide/virtual_machines/importing_unmanaging_vms.rst [hunk 2]
```diff
@@ -401,6 +409,170 @@ Unmanaging Instance actions
...
+MAC address that has been previously assigned by CloudStack. If it is set to true, the NICs with MAC addresses which
+already exist in the CloudStack database have the existing MAC addresses reassigned to its NICs.
```
Comment (DaanHoogland) [inline source/adminguide/virtual_machines/importing_unmanaging_vms.rst:509]: In general I think it makes sense to limit line lengths to about 80 chars , but I am not relegious about it;

```suggestion
.. note:: The `forced` parameter is false by default and prevents importing an Instance which has a NIC containing a
          MAC address that has been previously assigned by CloudStack. If it is set to true, the NICs with MAC addresses which
          already exist in the CloudStack database have the existing MAC addresses reassigned to its NICs.
```

File: source/adminguide/virtual_machines/importing_unmanaging_vms.rst [hunk 3]
```diff
@@ -45,7 +53,7 @@ replicated) and in the recreation of Instances which have been backed up (part o
 CloudStack's Backup and Recovery feature).
 
 The most complex part of importing Instances is the mapping of an unmanaged Instance's Networks to CloudStack Networks.  As an operator
```
Comment (andrijapanicsb) [inline source/adminguide/virtual_machines/importing_unmanaging_vms.rst:55]:
```suggestion
The most complex part of importing Instances is the mapping of an unmanaged Instance's Networks to CloudStack Networks, as an operator
```

**Comment by rohityadavcloud** [comment]
@blueorangutan docbuild

**Comment by rohityadavcloud** [comment]
@blueorangutan docbuild

**Comment by andrijapanicsb** [comment]
I've merged my changes into this PR - please re-review folks.

**Comment by andrijapanicsb** [comment]
@blueorangutan docbuild

File: source/adminguide/virtual_machines/importing_unmanaging_vms.rst [hunk 4]
```diff
@@ -401,6 +411,166 @@ Unmanaging Instance actions
...
+An External Host refers to a host that is not managed by CloudStack. The "Import from external host" feature enables importing/migrating
+instances from these external hosts. This feature is available in both UI and API.
+
+Prerequisites
+~~~~~~~~~~~~~
+- Ensure that the External KVM host is equipped with the installation of libvirt.
+- Allow TCP libvirt connections (listen_tpc=1) on those External Hosts from CloudStack hosts.
+- It is imperative that instances residing on the external host are in a stopped state, as live migration of instances is not supported
+- For some guest operating systems, it's also required that the operating system inside the Instance has been gracefully shut down.
+- Currently, it's supported to only use NFS and Local storage as the destination Primary Storage pools in CloudStack
```
Comment (andrijapanicsb) [inline source/adminguide/virtual_machines/importing_unmanaging_vms.rst:430]:
```suggestion
- Ensure that the External KVM host are running libvirt
- Allow libvirt TCP connections (listen_tpc=1) on those External Hosts from CloudStack hosts.
- Instances on the external host have to be in a stopped state, as live migration of instances is not supported
- For some guest operating systems, it's also required that the operating system inside the Instance has been gracefully shut down.
- Currently, it's supported to only use NFS and Local storage as the destination Primary Storage pools in CloudStack
- Currently, only libvirt-based instances can be migrated
```

**Comment by nvazquez** [comment]
@blueorangutan docbuild

**Review by shwstppr — APPROVED** [review]

File: source/adminguide/virtual_machines/importing_unmanaging_vms.rst [hunk 5]
```diff
@@ -39,37 +49,37 @@ Importing Unmanaged Instances
...
+If the 'destination' Network's VLAN(s) and the requested service offerings match the existing VLAN and the CPU/RAM profile of the Instance on the hypervisor level, then the Instance can be
+imported while it is running. If the VLANs or service offerings do not match, then the Instance to be imported must be stopped.
+Once the Instance has been added to CloudStack, starting it through CloudStack will alter the Instance's settings on the hypervisor in line with
+those set in the CloudStack DB (e.g. the Instance might be moved to a different Port Group on vSwitch/dvSwitch, with the corresponding VLAN)
 
 To import Instances, it is imagined that a Cloud Provider will:
 
-#. List all of the existing Networks which the Instances to be imported are on.
-#. Create corresponding Networks in CloudStack
+#. List/get familiar all of the existing Networks on which the Instances to be imported are on.
```
Comment (DaanHoogland) [inline source/adminguide/virtual_machines/importing_unmanaging_vms.rst:66]:
```suggestion
#. List/get familiar with all of the existing Networks on which the Instances to be imported are on.
```

**Review by DaanHoogland — COMMENTED** [review]
minor issue

**Comment by DaanHoogland** [comment]
I actually think we should merge this and fix forward

**Comment by shwstppr** [comment]
@DaanHoogland that works for me. @GutoVeronezi may add a new PR when feasible

**Comment by nvazquez** [comment]
@blueorangutan docbuild

**Review by nvazquez — APPROVED** [review]

**Pull request merged**